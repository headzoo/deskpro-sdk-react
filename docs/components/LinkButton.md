LinkButton
==========
A button which dispatches to a route location when clicked.

## Props

```jsx
<LinkButton
    to={string}
    params={object}
/>
```

## Examples

```jsx
<LinkButton to="settings">
    Click
</LinkButton>
<LinkButton to="settings" params={{ country: 'uk' }}>
    Click
</LinkButton>
```
