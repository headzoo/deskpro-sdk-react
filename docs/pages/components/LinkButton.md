Overview
========
The `LinkButton` component changes the current route location when clicked. See the [route documentation](/pages/props/#route) for more information.

## Props

```jsx
<LinkButton
    to={string}
    params={object}
/>
```

## Example
The following example renders two buttons which change the current location when clicked.

```jsx
import React from 'react';
import { LinkButton, sdkConnect } from 'deskpro-sdk-react';

class Page extends React.Component {
    render() {
        return (
            <div>
                <LinkButton to="settings">
                    Click
                </Link>
                <LinkButton to="settings" params={{ country: 'uk' }}>
                    Click
                </Link>
            </div>
        );
    }
}

export default sdkConnect(Page);
```

!!! note
    Components using the `LinkButton` component must be [connected to the SDK](/pages/props/#connecting-your-components) using `sdkConnect` or `DeskproSDK`.
