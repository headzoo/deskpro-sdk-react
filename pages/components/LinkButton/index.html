<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">   
    <link rel="shortcut icon" href="../../../favicon.png">

    <title>LinkButton - DeskPRO React SDK</title>

    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.0.3/css/bootstrap.min.css">
    <link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
    <link rel="stylesheet" href="//cdn.jsdelivr.net/font-hack/2.018/css/hack.min.css">
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto">
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.5.0/styles/agate.min.css">
    <link rel="stylesheet" href="../../../css/base.css">
    <link rel="stylesheet" href="../../../css/psinder.css">
    
    


    <link href="../../../css/site.css" rel="stylesheet">


    <link href="../../../css/admonition.css" rel="stylesheet">

    <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
            <script src="//oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="//oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
    <![endif]-->

    
</head>

<body>

    <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="../../..">
                <img src="../../../logo.png" />
                DeskPRO React SDK
            </a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../../..">Overview</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../manifest/">Manifest</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../props/">Props</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Components <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../DeskproSDK/">DeskproSDK</a>
</li>

                        
                            
<li >
    <a href="../Routes/">Routes</a>
</li>

                        
                            
<li >
    <a href="../Link/">Link</a>
</li>

                        
                            
<li class="active">
    <a href="./">LinkButton</a>
</li>

                        
                        </ul>
                    </li>
                
                
                
                    <li class="dropdown">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tutorials <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../../tutorials/basic/">Basic App</a>
</li>

                        
                            
<li >
    <a href="../../tutorials/form/">Form Handling</a>
</li>

                        
                            
<li >
    <a href="../../tutorials/oauth/">Using OAuth</a>
</li>

                        
                            
<li >
    <a href="../../tutorials/redux/">Using Redux</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li>
                        <a href="https://github.com/DeskproApps/sdk-react">
                            
                                <i class="fa fa-github"></i>
                            
                            GitHub
                        </a>
                    </li>
                
            </ul>
        </div>
    </div>
</div>

    <div class="container">
        
        <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="first-level active"><a href="#overview">Overview</a></li>
        
            <li class="second-level"><a href="#props">Props</a></li>
            
        
            <li class="second-level"><a href="#example">Example</a></li>
            
        
    
    </ul>
</div></div>
        <div class="col-md-9" role="main">

<h1 id="overview">Overview</h1>
<p>The <code>LinkButton</code> component changes the current route location when clicked. See the <a href="../../../pages/props/#route">route documentation</a> for more information.</p>
<h2 id="props">Props</h2>
<pre><code class="jsx">&lt;LinkButton
    to={string}
    params={object}
/&gt;
</code></pre>

<h2 id="example">Example</h2>
<p>The following example renders two buttons which change the current location when clicked.</p>
<pre><code class="jsx">import React from 'react';
import { LinkButton, sdkConnect } from 'deskpro-sdk-react';

class Page extends React.Component {
    render() {
        return (
            &lt;div&gt;
                &lt;LinkButton to=&quot;settings&quot;&gt;
                    Click
                &lt;/Link&gt;
                &lt;LinkButton to=&quot;settings&quot; params={{ country: 'uk' }}&gt;
                    Click
                &lt;/Link&gt;
            &lt;/div&gt;
        );
    }
}

export default sdkConnect(Page);
</code></pre>

<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Components using the <code>LinkButton</code> component must be <a href="../../../pages/props/#connecting-your-components">connected to the SDK</a> using <code>sdkConnect</code> or <code>DeskproSDK</code>.</p>
</div></div>
        
    </div>

    <footer class="col-md-12 text-center">
        
            
            <p><a href="https://github.com/DeskproApps/sdk-react/tree/master/docs/pages/components/LinkButton.md" class="icon icon-github"> Edit this page on GitHub</a></p>
            
        
        
            <small>Copyright 2001 â€“ 2017 Deskpro Ltd. All rights reserved.<br></small>
        
    </footer>

    <script src="//code.jquery.com/jquery-2.2.4.min.js"></script>
    <script src="//maxcdn.bootstrapcdn.com/bootstrap/3.0.3/js/bootstrap.min.js"></script>
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
    <script>
    var base_url = '../../..';
    </script>
    <script data-main="../../../mkdocs/js/search.js" src="../../../mkdocs/js/require.js"></script>
    <script src="../../../js/base.js"></script>

    <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
        <div class="modal-dialog">
            <div class="modal-content">
                <div class="modal-header">
                    <button type="button" class="close" data-dismiss="modal">
                        <span aria-hidden="true">&times;</span>
                        <span class="sr-only">Close</span>
                    </button>
                    <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                </div>
                <div class="modal-body">
                    <p>
                        From here you can search these documents. Enter your search terms below.
                    </p>
                    <form role="form">
                        <div class="form-group">
                            <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                        </div>
                    </form>
                    <div id="mkdocs-search-results"></div>
                </div>
            </div>
        </div>
    </div>
    </body>
</html>
