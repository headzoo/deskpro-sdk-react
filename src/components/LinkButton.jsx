import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'deskpro-components';
import { propKeyFilter } from '../utils/props';

/**
 * A button which dispatches to a route location when clicked.
 *
 * Example:
 *
 * ```jsx
 * render() {
 *  return (
 *    <div>
 *      <LinkButton to="index">Index Page</LinkButton>
 *    </div>
 *  );
 * }
 * ```
 */
class LinkButton extends React.Component {
  static propTypes = {
    /**
     * Route that will be dispatched when the button is clicked.
     */
    to: PropTypes.string,

    /**
     * Params passed with the route.
     */
    params: PropTypes.object,

    /**
     * Children rendered into the button.
     */
    children: PropTypes.node,

    /**
     * Called when the button is clicked.
     */
    onClick: PropTypes.func
  };

  /**
   * Specifies the default values for props
   */
  static defaultProps = {
    to:       '',
    params:   {},
    children: '',
    onClick:  () => {}
  };

  /**
   * Context values the child wants passed down from the parent
   */
  static contextTypes = {
    route: PropTypes.object
  };

  /**
   * Called when the button is clicked
   *
   * Dispatches to the 'to' route unless the default is prevented.
   *
   * @param {Event} e
   */
  handleClick = (e) => {
    this.props.onClick(e);
    if (!e.defaultPrevented) {
      this.context.route.to(this.props.to, this.props.params);
    }
  };

  /**
   * @returns {XML}
   */
  render() {
    const { children, ...props } = this.props;

    return (
      <Button {...propKeyFilter(props, LinkButton.propTypes)} onClick={this.handleClick}>
        {children}
      </Button>
    );
  }
}

export default LinkButton;
