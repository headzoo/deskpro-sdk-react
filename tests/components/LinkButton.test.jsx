import React from 'react';
import renderer from 'react-test-renderer';
import LinkButton from '../../src/components/LinkButton';

it('LinkButton renders a button', () => {
  const renderedValue = renderer.create(
    <LinkButton to="index">Index</LinkButton>
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
