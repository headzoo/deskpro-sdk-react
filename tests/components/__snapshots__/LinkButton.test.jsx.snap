// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders a button 1`] = `
<button
  className="dp-button dp-button--l dp-button--primary dp-button--shape-default"
  disabled={false}
  onClick={[Function]}
>
  Index
</button>
`;
